.evergreen/compile-unix.sh

```diff
@@ -228,7 +228,7 @@ pkg-config --modversion libssl || true
 if [ "$COMPILE_LIBMONGOCRYPT" = "ON" ]; then
    # Build libmongocrypt, using the previously fetched installed source.
    # TODO (CDRIVER-4397): add "--branch 1.5.0-rc0" in git clone once libmongocrypt 1.5.0-rc0 is released.
-   git clone https://github.com/mongodb/libmongocrypt
+   git clone https://github.com/kevinAlbs/libmongocrypt --branch fix-leak
```

Comment:
TODO: revert this change and the change in compile-windows.sh once https://github.com/mongodb/libmongocrypt/pull/357 is merged.